ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/JdbcLedgerDao.scala

```diff
@@ -604,14 +597,36 @@ private class JdbcLedgerDao(
   }
 
   override def prune(
-      pruneUpToInclusive: Offset
-  )(implicit loggingContext: LoggingContext): Future[Unit] =
-    dbDispatcher.executeSql(metrics.daml.index.db.pruneDbMetrics) { conn =>
-      storageBackend.pruneEvents(pruneUpToInclusive)(conn)
-      storageBackend.pruneCompletions(pruneUpToInclusive)(conn)
-      storageBackend.updatePrunedUptoInclusive(pruneUpToInclusive)(conn)
-      logger.info(s"Pruned ledger api server index db up to ${pruneUpToInclusive.toHexString}")
-    }
+      pruneUpToInclusive: Offset,
+      pruneAllDivulgedContracts: Boolean,
+  )(implicit loggingContext: LoggingContext): Future[Unit] = {
+    val allDivulgencePruningParticle =
+      if (pruneAllDivulgedContracts) " (including all divulged contracts)" else ""
+    logger.info(s"Starting pruning of ledger api server index db$allDivulgencePruningParticle")
+
+    dbDispatcher
+      .executeSql(metrics.daml.index.db.pruneDbMetrics) { conn =>
+        storageBackend.pruneEvents(pruneUpToInclusive, pruneAllDivulgedContracts)(
+          conn,
+          loggingContext,
+        )
+
+        storageBackend.pruneCompletions(pruneUpToInclusive)(conn, loggingContext)
+        storageBackend.updatePrunedUptoInclusive(pruneUpToInclusive)(conn)
+
+        if (pruneAllDivulgedContracts) {
+          storageBackend.updatePrunedAllDivulgedContractsUpToInclusive(pruneUpToInclusive)(conn)
+        }
+      }
+      .andThen {
+        case Success(_) =>
+          logger.info(
+            s"Completed pruning of the ledger api server index db$allDivulgencePruningParticle up to ${pruneUpToInclusive.toHexString}"
+          )
+        case Failure(ex) =>
+          logger.warn("Pruning failed", ex)
```

Comment:
I would change the logging around pruning somewhat:
- include the pruneUpToInclusive in the first log at the entry to the function
- Rephrase the initial messge to: "Pruning the ledger api server index db$allDivulgencePruningParticle up to ${pruneUpToInclusive.toHexString}"
- Shorten the successful completion to just "Pruning succeeded"

Reply:
Done